raiden/blockchain/events.py

```diff
@@ -256,10 +256,23 @@ class BlockchainEvents:
 
     def __init__(self):
         self.event_listeners = list()
+        self.first_run = True
+
+    def reset(self):
+        self.first_run = True
 
     def poll_blockchain_events(self):
+        # When we test with geth if the contracts have already been deployed
+        # before the filter creation we need to use `get_all_entries` to make
+        # sure we get all the events. With tester this is not required.
```

Comment:
why? aren't the filters installed since block 0?

Reply:
So this one is a good question. I can't answer it yet. We will need to look this up further. Already trying to ask geth people about this. I also asked in the standup. The problem is that even if the filters are installed since block 0, if the contracts were deployed before the filter installation it seems the events are not returned with `get_new_events` but only with `get_all_events`.